Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/review.py

```diff
@@ -0,0 +1,235 @@
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import difflib
+import os
+import re
+import sys
+import tempfile
+
+from .command import Command
+from .pull_request import PullRequest
+
+from webkitcorepy import arguments, run, string_utils, Terminal
+from webkitscmpy import local, log, remote
+
+
+class Review(Command):
+    name = 'review'
+    help = "Run a command-line wizard to review a specific pull-request or commit in a project"
+    DETAILS_RE = r'<details>.+</details>'
+    INDENT_SIZE = 4
+    PR_RE = re.compile(r'^\[?([Pp][Rr][ -]?)?(?P<number>\d+)]?$')
+    PR_URL_RES = [
+        re.compile(r'\A(?P<url>https?://\S+/projects/\S+/repos/\S+)/pull-requests/(?P<number>\d+)(/overview)?(/diff)?\Z'),
+        re.compile(r'\A(?P<url>https?://github.\S+/\S+)/pull/(?P<number>\d+)(/files)?\Z'),
+    ]
+    SEPERATOR_SIZE = 80
+    URL_RE = re.compile(r'\Ahttps?://')
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        parser.add_argument(
+            'argument', nargs=1,
+            type=str, default=None,
+            help='String representation of a pull request (pr-#), commit or URL to review',
+        )
+        parser.add_argument(
+            '--remote', dest='remote', type=str, default=None,
+            help='Specify remote to search for pull request from.',
+        )
+        parser.add_argument(
+            '--dry-run', '-d', '--no-dry-run',
+            dest='dry_run', default=False,
+            help='View but do not edit a pull-request',
+            action=arguments.NoAction,
+        )
+
+    @classmethod
+    def editor(cls, repository):
+        from_config = None
+        if isinstance(repository, local.Git):
+            from_config = repository.config().get('core.editor', None)
+        if not from_config:
+            from_config = local.Git.config().get('core.editor', None)
+        if from_config:
+            return from_config.split(' ')
+
+        from whichcraft import which
+        return [which('vi')]
```

Comment:
nit: On some linux distributions (iirc, anything Debian-based), `vi` will be the ancient POSIX vi program, not vim. So you might want to default to vim?